pkg/virt-handler/vm.go

```diff
@@ -499,6 +508,30 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 		condManager.RemoveCondition(vmi, v1.VirtualMachineInstanceAgentConnected)
 	}
 
+	// sync vmi guest time if aganet is availble and remove the required time sync condition
+	if condManager.HasCondition(vmi, v1.VirtualMachineInstanceRequireTimeSync) {
+		timeSyncRequestTimeoutInSeconds := 5.0
+		isConditionRemoved := false
+		if channelConnected {
+			err := d.setVMIGuestTime(vmi)
+			if err == nil {
+				condManager.RemoveCondition(vmi, v1.VirtualMachineInstanceRequireTimeSync)
```

Comment:
I think you want to set isConditionRemoved to true here

Reply:
yep, thanks